kits/arm/ex_AR_kit.cpp

```diff
@@ -0,0 +1,223 @@
+  std::vector<std::string> errors;
+  
+  // Load the config
+  const auto example_config = RobotConfig::loadConfig(example_config_file, errors);
+  for (const auto& error : errors) {
+    std::cerr << error << std::endl;
+  }
+  if (!example_config) {
+    std::cerr << "Failed to load configuration from: " << example_config_file << std::endl;
+    return -1;
+  }
+
+  // For this demo, we need the arm and mobile_io
+  std::unique_ptr<hebi::experimental::arm::Arm> arm;
+  std::unique_ptr<hebi::util::MobileIO> mobile_io;
+
+  //////////////////////////
+  ///// Arm Setup //////////
+  //////////////////////////
+
+  // Create the arm object from the configuration
+  arm = hebi::experimental::arm::Arm::create(*example_config);
+
+  // Keep retrying if arm not found
+  while (!arm) {
+      std::cerr << "Failed to create arm, retrying..." << std::endl;
+
+      // Wait for 1 second before retrying
+      std::this_thread::sleep_for(std::chrono::seconds(1));  
+
+      // Retry
+      arm = hebi::experimental::arm::Arm::create(*example_config);
+  }
+  std::cout << "Arm connected." << std::endl;
+
+  //////////////////////////
+  //// MobileIO Setup //////
+  //////////////////////////
+
+  // Create the mobile_io object from the configuration
+  std::cout << "Waiting for Mobile IO device to come online..." << std::endl;
+  mobile_io = createMobileIOFromConfig(*example_config, example_config_file);
+
+  // Keep retrying if Mobile IO not found
+  while (mobile_io == nullptr) {
+      std::cout << "Couldn't find Mobile IO. Check name, family, or device status..." << std::endl;
+
+      // Wait for 1 second before retrying
+      std::this_thread::sleep_for(std::chrono::seconds(1));  
+
+      // Retry
+      mobile_io = createMobileIOFromConfig(*example_config, example_config_file);
+  }
+  std::cout << "Mobile IO connected." << std::endl;
+
+  // Setup instructions for display
+  std::string instructions;
+  instructions = ("🏠 - Home Position\n📲 - AR Control Mode\n"
+                  "🌍 - Grav Comp Mode\n❌ - Quit\n");
+
+  // Clear any garbage on screen
+  mobile_io->clearText();
+
+  // Display instructions on screen
+  mobile_io->appendText(instructions); 
+
+  // Setup state variable for mobile device
+  auto last_mobile_state = mobile_io->update();
+
+  //////////////////////////
+  //// Main Control Loop ///
+  //////////////////////////
+
+  // Different modes it can be in (when in none, then automatic grav_comp)
+  bool softstart = true;
+  bool ar_mode = false;
+  
+  // Load user data from config
+  Eigen::VectorXd home_position(arm -> robotModel().getDoFCount());
+  home_position = Eigen::Map<const Eigen::VectorXd>(example_config->getUserData().float_lists_.at("home_position").data(), example_config->getUserData().float_lists_.at("home_position").size());  
+  double soft_start_time = example_config->getUserData().floats_.at("soft_start_time");
+  double xyz_scale = example_config->getUserData().floats_.at("xyz_scale");
```

Comment:
Looking at the usage here, I'm thinking it might be clearer to have this look like:
`double xyz_scale = example_config->getUserData().getFloats("xyz_scale");`

This way, the syntax looks a little more readable (imo), and also can be helpful for debugging to have breakpoints within a function (whereas this isn't as straightforward with direct member access).

Reply:
Done